builtin/logical/ssh/path_issuers.go

```diff
@@ -0,0 +1,517 @@
+		Default:     false,
+	}
+
+	return &framework.Path{
+		Pattern: "issuers/import" + framework.OptionalGenericNameRegex("issuer_name"),
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixSSH,
+			OperationVerb:   "submit",
+			OperationSuffix: "issuer",
+		},
+
+		Fields: fields,
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathWriteIssuerHandler,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationVerb:   "submit",
+					OperationSuffix: "issuer",
+				},
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {
+						{
+							Description: "OK",
+							Fields:      issuerOKResponseFields,
+						},
+					},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathSubmitIssuerSyn,
+		HelpDescription: pathSubmitIssuerDesc,
+	}
+}
+
+func pathListIssuers(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issuers/?$",
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixSSH,
+			OperationVerb:   "list",
+			OperationSuffix: "issuers",
+		},
+
+		Fields: map[string]*framework.FieldSchema{
+			"after": {
+				Type:        framework.TypeString,
+				Required:    false,
+				Description: `Optional entry to list begin listing after, not required to exist.`,
+			},
+			"limit": {
+				Type:        framework.TypeInt,
+				Required:    false,
+				Description: `Optional number of entries to return; defaults to all entries.`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback: b.pathListIssuersHandler,
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {{
+						Description: "OK",
+						Fields: map[string]*framework.FieldSchema{
+							"keys": {
+								Type:        framework.TypeStringSlice,
+								Description: `A list of keys`,
+								Required:    true,
+							},
+							"key_info": {
+								Type:        framework.TypeMap,
+								Description: `Key info with issuer identifier`,
+								Required:    false,
+							},
+						},
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathListIssuersSyn,
+		HelpDescription: pathListIssuersDesc,
+	}
+}
+
+func pathGetIssuerPublicKeyUnauthenticated(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issuer/" + framework.GenericNameRegex("issuer_ref") + "/public_key",
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixSSH,
+			OperationVerb:   "get",
+			OperationSuffix: "issuer",
+		},
+
+		Fields: map[string]*framework.FieldSchema{
+			"issuer_ref": {
+				Type:        framework.TypeString,
+				Description: `Issuer reference. It can be the issuer's unique identifier, or the optionally given name.`,
```

Comment:
This might apply in other places, but can't it also take the value `default`?

Reply:
It should as, If I'm not mistaken and going to confirm, we always use `resolveIssuerReference` to resolve issuer references provided. And this function does handle `default` being passed.

```
	if ref == defaultRef {
		// If reference is 'default', fetch the default issuer ID from the configuration
		issuerConfig, err := sc.getIssuersConfig()
		if err != nil {
			return "", err
		}
		if len(issuerConfig.DefaultIssuerID) == 0 {
			return IssuerRefNotFound, errutil.UserError{Err: "no default issuer currently configured"}
		}

		return issuerConfig.DefaultIssuerID, nil
	}
...
```

Reply:
Perfect, want to add that to this description? Thanks!

Reply:
Yes, added. Just haven't pushed the commit yet.